### ISSUE TITLE
Real world interfaces

### ISSUE THREAD

**Issue by Noitarud** [phyphox/phyphox-android#55]
There are teachers embracing robotics, I presume you m.i.g.h.t have an interest in this… I do not know if android is compatible with old school technology (see image) but the sensors of device could be used as triggers for operating motors or heaters, to make responses or corrections.

Image shows an OTG cable (useful with Android devices) with a USB-LPT(printer) adaptor and an experimentation board for Parallel port. I was sure the serial port had some I/O too.
![IMG_20230513_213821](https://github.com/phyphox/phyphox-android/assets/98794719/a486f70b-c224-44f0-9b89-5242d2119227)

**Comment by Staacks** [comment]
Without having tested it myself, I am quite sure that Android can handle data inputs via OTG cables (or just acting as a USB host in case of modern USB-C cables). However, there are two problems which prevented us from implementing this into phyphox so far:
- As far as I can tell, this is not possible on Apple devices. This limits the number of student devices that can be used in teaching with this interface and makes it impossible to use it on most school-provided sets of devices, which are typically iPads (at least in Germany). This does not prevent the feature from being included in phyphox, but moves it from our todo list to our nice-to-have list.
- There is no universal USB protocol for such devices. Most of them will probably use the USB CDC protocol to implement a serial communication over USB, which in turn means that we need to implement support for such devices in an extremely generic way, so that different serial protocols are supported. (For example streaming binary values, streaming individual ASCII encoded descimal numbers, streaming comma seperated numbers, extracting number from a string like "T = 20.7C", JSON-encoded data packets etc. - or even needing phyphox to request data with specific commands)
- Most of these USB-sensors are DIY devices and use a serial interface. Usually, this means that it is not too difficult to attach them to an ESP32 and connect them to phyphox through our Arduino library. So, many of these DIY sensors have a good alternative  to USB or are shifting to the Arduino (or MicroPython) ecosystem anyway.

In summary, this does not rule out that phyphox could support USB data sources eventually, but it is not a priority and will probably not be implemented by us in the foreseeable future. (Pull requests are always welcome, of course.)

**Comment by Staacks** [comment]
Currently reviewing open issues and three years later I would still say that this answer is the same. So, I am closing this as "not planned". Sorry.
